sycl/test-e2e/E2EExpr.py

```diff
@@ -0,0 +1,219 @@
+                "!(windows || zstd)", {"linux"}, True, final_unknown_value=False
+            )
+        )
+        # build-only expressions where unknown affects the resulting value
+        self.assertTrue(
+            E2EExpr.evaluate("rt_feature", {}, True, final_unknown_value=True)
+        )
+        self.assertFalse(
+            E2EExpr.evaluate("rt_feature", {}, True, final_unknown_value=False)
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "rt_feature", {"rt_feature"}, True, final_unknown_value=True
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "rt_feature", {"rt_feature"}, True, final_unknown_value=False
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate("!rt_feature", {}, True, final_unknown_value=False)
+        )
+        self.assertFalse(
+            E2EExpr.evaluate("!!rt_feature", {}, True, final_unknown_value=False)
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "windows || rt_feature", {"linux"}, True, final_unknown_value=True
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "windows || rt_feature", {"linux"}, True, final_unknown_value=False
+            )
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "linux && rt_feature", {"linux"}, True, final_unknown_value=True
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "linux && rt_feature", {"linux"}, True, final_unknown_value=False
+            )
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "linux && !(windows || rt_feature)",
+                {"linux"},
+                True,
+                final_unknown_value=True,
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "linux && !(windows || rt_feature)",
+                {"linux"},
+                True,
+                final_unknown_value=False,
+            )
+        )
+        # build-only expressions where unknown does not affect the resulting value
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "linux || rt_feature", {"linux"}, True, final_unknown_value=True
+            )
+        )
+        self.assertTrue(
+            E2EExpr.evaluate(
+                "linux || rt_feature", {"linux"}, True, final_unknown_value=False
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "windows && rt_feature", {"linux"}, True, final_unknown_value=True
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "windows && rt_feature", {"linux"}, True, final_unknown_value=False
+            )
+        )
+        self.assertFalse(
+            E2EExpr.evaluate(
+                "linux && (windows && rt_feature)",
```

Comment:
change "windows" to something that can make sense in real world. Same in the next one.